### PR TITLE
Github Improvement: add Confluence Code-Review Guideline Page

### PR DESCRIPTION
**Pull request by jamalxo**
Added Code-Review Guidelines from Confluence

### CODE REVIEW

**Review by NicolasUllmann — COMMENTED** [review]
Checked for typos. Looks good.

**Review by willergeorg — COMMENTED** [review]
In general looks good, just one bullet point @NicolasUllmann mentioned needs some correction. And try to keep consistent between the bullet points whether to end with or without a punctuation mark.

File: docs/dev/guidelines/code-review.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
+******
+Code Review
```
Comment (FrankeLukas) [inline docs/dev/guidelines/code-review.rst:2]: This file is not referenced in `guidelines.rst`. GitHub actions also complains about several title underlines being too short (go to the changes view of this PR to see the complaints).

**Review by FrankeLukas — CHANGES_REQUESTED** [review]

File: docs/dev/guidelines/code-review.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+* Verify that related functionality is still working as expected.
+* Check the Changes to
+* Conform with the code style
+* Make sure you can easily understand the code
+* Make sure that (extensive) comments are present where deemed necessary
+* Performance is reasonable (e.g. Number of database queries or HTTP calls)
+* Submit your comments and status (Approve or Request Changes) using GitHub.
+
+4. Respond to review- > Developer & Reviewer
+=====================
```
Comment (ivanchimeno) [inline docs/dev/guidelines/code-review.rst:42]: You missed a codacy issue here. I think you need to underline the title completely.

File: docs/dev/guidelines/code-review.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+0. Precondition -> only Developer
+==================================
+
+* Limit yourself to one functionality per pull request.
+* Split up your task in multiple branches & pull requests if necessary.
+* `Commit Early, Commit Often, Perfect Later, Publish Once <https://speakerdeck.com/lemiorhan/10-git-anti-patterns-you-should-be-aware-of>`_.
+
+1. Start Implementation -> only Developer
+========================================
```
Comment (ivanchimeno) [inline docs/dev/guidelines/code-review.rst:16]: You missed a codacy issue here. I think you need to add one equals character

**Review by ivanchimeno — CHANGES_REQUESTED** [review]
You missed some Codacy warnings :)

File: docs/dev/guidelines/code-review.rst (hunk 1 above)
Comment (jamalxo) [reply docs/dev/guidelines/code-review.rst:2]: thank you, fixed

File: docs/dev/guidelines/code-review.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+* `Open draft pull request. <https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request>`_ Allows for code related questions and discussions
+
+2. Implementation is "done" -> only Developer
+=============================================
+
+* Make sure all steps in the `Checklist <https://github.com/ls1intum/ArTEMiS/blob/develop/.github/PULL_REQUEST_TEMPLATE.md>`_ are completed
+* Add or update the "Steps for Testing" in the description of your pull request
+* Mark the pull request as `ready for review <https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request>`_
+* Add Stephan Krusche (@krusche) and Jan Philip Bernius (@jpbernius) and your review partner as reviewers to the pull request
```
Comment (FrankeLukas) [inline docs/dev/guidelines/code-review.rst:26]: `@krusche` and `@jpbernius` seem to be added automatically already, so not sure if we have to list that here too.

File: docs/dev/guidelines/code-review.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+STEPS
+======
+
+0. Precondition -> only Developer
+==================================
+
+* Limit yourself to one functionality per pull request
+* Split up your task in multiple branches & pull requests if necessary
+* `Commit Early, Commit Often, Perfect Later, Publish Once <https://speakerdeck.com/lemiorhan/10-git-anti-patterns-you-should-be-aware-of>`_
```
Comment (FrankeLukas) [inline docs/dev/guidelines/code-review.rst:13]: Imho an article would be better than a slide show with mostly images, but I don't have one to suggest at hand.

File: docs/dev/guidelines/code-review.rst [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+* Perform the "Steps for Testing" and verify that the new functionality is working as expected
+* Verify that related functionality is still working as expected
+* Check the Changes to conform with the code style
+* Make sure you can easily understand the code
+* Make sure that (extensive) comments are present where deemed necessary
+* Performance is reasonable (e.g. Number of database queries or HTTP calls)
+* Submit your comments and status (Approve or Request Changes) using GitHub
+
+4. Respond to review- > Developer & Reviewer
```
Comment (FrankeLukas) [inline docs/dev/guidelines/code-review.rst:40]:
```suggestion
4. Respond to review -> Developer & Reviewer
```

**Review by FrankeLukas — COMMENTED** [review]
Looks good to me generally. I find the `-> only ...` a bit odd, and think that could be moved out of the headline. That is just my opinion though.

File: docs/dev/guidelines/code-review.rst (hunk 5 above)
Comment (jamalxo) [reply docs/dev/guidelines/code-review.rst:13]: This is just based from the Confluence page, but yea we could improve this

File: docs/dev/guidelines/code-review.rst (hunk 4 above)
Comment (jamalxo) [reply docs/dev/guidelines/code-review.rst:26]: yes, we should remove this point

File: docs/dev/guidelines/code-review.rst (hunk 6 above)
Comment (jamalxo) [reply docs/dev/guidelines/code-review.rst:40]: fixed

**Review by ivanchimeno — APPROVED** [review]
Looks good to me now

**Review by FrankeLukas — APPROVED** [review]

**Review by NicolasUllmann — APPROVED** [review]

**Pull request merged**